src/hard.js

```diff
@@ -0,0 +1,90 @@
+function fizzBuzz(start, end) {
+  let fizzBuzz = [];
+
+  for (let i = start; i < end; i++) {
+    if (i % 15 === 0) {
+      fizzBuzz.push("fizzbuzz");
+    } else if (i % 3 === 0) {
+      fizzBuzz.push("fizz");
+    } else if (i % 5 === 0) {
+      fizzBuzz.push("buzz");
+    } else {
+      fizzBuzz.push(i);
+    }
+  }
+  return fizzBuzz;
+}
+
+function hasDuplicate(numbers) {
+  let numberTally = {};
+
+  for (let number = 0; number < numbers.length; number++) {
+    if (numbers[number] in numberTally) {
+      return true;
+    } else {
+      numberTally[numbers[number]] = 1;
+    }
+  }
+
+  return false;
+}
+
+function findPlurality(items) {
+  if (items.length === 0) {
+    return undefined;
+  }
+
+  let itemCounter = {};
+
+  for (let i = 0; i < items.length; i++) {
+    if (items[i] in itemCounter) {
+      itemCounter[items[i]]++;
+    } else {
+      itemCounter[items[i]] = 1;
+    }
+  }
+
+  let maxCount = 0;
+
+  for (const item in itemCounter) {
+    if (itemCounter[item] > maxCount) {
+      maxCount = itemCounter[item];
+    }
```

Comment:
we can do a nice trick here:

```suggestion
    maxCount = Math.max(itemCounter[item], maxCount);
```